fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -612,6 +612,309 @@ void partitionedJoinedIndexTest() {
+
+            validateSegmentAndIndexIntegrity(COMPLEX_PARTITIONED, partition0Subspace, context, "_0.cfs");
+            validateSegmentAndIndexIntegrity(COMPLEX_PARTITIONED, partition1Subspace, context, "_0.cfs");
+        }
+    }
+
+    /**
+     * test cross partition limit query with multiple scans.
+     */
+    @Test
+    void testPartitionedLimitNeedsMultipleScans() {
+        try (FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, COMPLEX_DOC, COMPLEX_PARTITIONED);
+
+            long timestamp60DaysAgo = Instant.now().minus(60, ChronoUnit.DAYS).toEpochMilli();
+            long timestamp29DaysAgo = Instant.now().minus(29, ChronoUnit.DAYS).toEpochMilli();
+
+            // partition 0 covers 30 days beginning 60 days ago
+            createPartitionMetadata(COMPLEX_PARTITIONED, Tuple.from(1), 0,
+                    timestamp60DaysAgo,
+                    Instant.now().minus(30, ChronoUnit.DAYS).toEpochMilli());
+
+            // partition 1 covers the past 29 days till yesterday
+            createPartitionMetadata(COMPLEX_PARTITIONED, Tuple.from(1), 1,
+                    timestamp29DaysAgo,
+                    Instant.now().minus(1, ChronoUnit.DAYS).toEpochMilli());
+
+            // 30 records in partition 0
+            for (int i = 0; i < 300; i++) {
+                recordStore.saveRecord(createComplexDocument(1111L + i, ENGINEER_JOKE, 1, Instant.now().minus(40, ChronoUnit.DAYS).toEpochMilli()));
+            }
+
+            // 30 records in partition 1
+            for (int i = 0; i < 300; i++) {
+                recordStore.saveRecord(createComplexDocument(2222L + i, ENGINEER_JOKE, 1, Instant.now().minus(20, ChronoUnit.DAYS).toEpochMilli()));
+            }
+
+            assertEquals(451, recordStore.scanIndex(COMPLEX_PARTITIONED, groupedTextSearch(COMPLEX_PARTITIONED, "text:propose", 1), null, ExecuteProperties.newBuilder().setReturnedRowLimit(451).build().asScanProperties(false))
+                    .getCount().join());
+            assertEquals(3, getCounter(context, LuceneEvents.Events.LUCENE_INDEX_SCAN).getCount());
+            assertEquals(451, getCounter(context, LuceneEvents.Counts.LUCENE_SCAN_MATCHED_DOCUMENTS).getCount());
+
+            final Subspace partition0Subspace = recordStore.indexSubspace(COMPLEX_PARTITIONED).subspace(Tuple.from(1, LucenePartitioner.PARTITION_DATA_SUBSPACE).add(0));
+            final Subspace partition1Subspace = recordStore.indexSubspace(COMPLEX_PARTITIONED).subspace(Tuple.from(1, LucenePartitioner.PARTITION_DATA_SUBSPACE).add(1));
+
+            validateSegmentAndIndexIntegrity(COMPLEX_PARTITIONED, partition0Subspace, context, "_0.cfs");
+            validateSegmentAndIndexIntegrity(COMPLEX_PARTITIONED, partition1Subspace, context, "_0.cfs");
+        }
+    }
+
+    /**
+     * test cross partition skip over max page size.
+     */
+    @Test
+    void testPartitionedSkipOverMaxPageSize() {
```

Comment:
do you mean a test where we specify a relevance sort, but get results which aren't in a strict relevance order because of which partition they happen to be in?

Reply:
Yes, that is what I mean.
It would confirm our expectations of how crossing the partition boundaries will manifest.